tests/robot-tests/tests/libs/slack.py

```diff
@@ -0,0 +1,96 @@
+import os
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+import shutil
+import json
+import requests
+from bs4 import BeautifulSoup
+
+PATH = f'{os.getcwd()}{os.sep}test-results'
+
+
+def _generate_slack_attachments():
+    failed_tests = dict()
+    passed_tests = dict()
+    skipped_tests = dict()
+
+    report = open(f'{PATH}{os.sep}output.xml', 'rb')
+    contents = report.read()
+
+    soup = BeautifulSoup(contents, 'xml')
+    test = soup.find('total').find('stat')
+
+    failed_tests = test['fail']
+    passed_tests = test['pass']
+    skipped_tests = test['skip']
+
+    return [
+        {
+            "pretext": "All results",
+            "color": "danger" if failed_tests or skipped_tests is not None or not "" else "good",
+            "mrkdwn_in": [
+                "pretext"
+            ],
+            "fields": [
+                {
+                    "title": "Total test cases",
+                    "value": int(skipped_tests) + int(passed_tests) + int(failed_tests)
+                },
+                {
+                    "title": "Passed tests",
+                    "value": int(passed_tests)
+                },                
+                {
+                    "title": "Failed tests",
+                    "value": int(failed_tests) + int(skipped_tests) if failed_tests or skipped_tests is not None or not "" else "0"
```

Comment:
Why are skipped tests considered failing? This seems like a false positive to me

Reply:
Ah yes, that's a good catch. I initially thought that tests get marked as skipped if we fail them fast (due to the logic in https://github.com/dfe-analytical-services/explore-education-statistics/blob/dev/tests/robot-tests/listeners/KeywordListener.py)